src/Api/Serializer/RestSerializer.php

```diff
@@ -126,6 +126,15 @@ private function applyHeader($name, Shape $member, $value, array &$opts)
         if ($member->getType() == 'timestamp') {
             $value = TimestampShape::format($value, 'rfc822');
         }
+        if ($member['jsonvalue']) {
+            $value = json_encode($value);
+            if ($value === null && JSON_ERROR_NONE !== json_last_error()) {
+                throw new \ParserException('Unable to encode the provided value'
```

Comment:
If the value is a resource, `json_last_error()` will return `JSON_ERROR_UNSUPPORTED_TYPE`, but users could also encounter other errors specific to the act of encoding like `JSON_ERROR_DEPTH` or `JSON_ERROR_RECURSION`.

My point was that this PR could add a test to make sure providing invalid data for a jsonValue header throws a catchable exception. Prior to https://github.com/aws/aws-sdk-php/pull/1229/commits/93abe6baf75b7b73e1c2f7973c541fc5bb337d01, the code would have raised an uncatchable fatal error, but the tests were still green. The tests currently only exercise the happy path and not any expected error conditions.